src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -793,7 +793,7 @@ abstract class BTypes {
     )
     def unapply(cr:ClassBType) = Some(cr.internalName)
 
-    def apply(internalName: InternalName, fromSymbol: Boolean)(init: (ClassBType) => Either[NoClassBTypeInfo, ClassInfo]): ClassBType = {
+    final def apply[T](internalName: InternalName, t: T, fromSymbol: Boolean)(init: (ClassBType, T) => Either[NoClassBTypeInfo, ClassInfo]): ClassBType = {
```

Comment:
Maybe add a comment what `t: T` is for; "Allows hoisting the lambda argument at call-sites for allocation reduction, by passing a captured value as parameter"